ADAPT-5012 | @rebeccahongsf | fix up registration interstitial page style

rebeccahongsf opened the pull request:
# READY FOR REVIEW

# Summary
- update body text to plain textarea field
- add `su-subheading` to body textarea field
- update H3 to use `size={5}`
- center remove/add button with traveler name
- remove grid gap and add `margin-bottom: 18px` (excluding last child) to traveler cards
- update text above card list to use responsive spacing

# Review By (Date)
- before end of Sprint W

# Review Tasks

## Setup tasks and/or behavior to test

1. Navigate to [`/travel-study/destinations/italy-extension/register`](https://deploy-preview-407--stanford-alumni.netlify.app/travel-study/destinations/italy-extension/register)
2. Review styles and confirm that the request changes have been implemented
3. Review code

# Associated Issues and/or People
- ADAPT-5012

yvonnetangsu commented on src/components/page-types/registrationFormPage/interstitialPage.js:
```diff
@@ -187,14 +185,20 @@ const InterstitialPage = (props) => {
                   </GridCell>
                   <GridCell xs={12} md={7} lg={8}>
-                    <p className="su-intro-text">
+                    <p className="su-subheading su-mb-0 su-rs-mt-1">
```
My bad - I must have mistaken the previously rich text block to be this paragraph instead of the one above. This one has the original `su-intro-text` class instead of `su-subheading`. My apologies 🙇🏼‍♀️ 😹

yvonnetangsu commented on src/components/page-types/registrationFormPage/interstitialPage.js:
```diff
@@ -187,14 +185,20 @@ const InterstitialPage = (props) => {
                   </GridCell>
                 </Grid>
-                <Grid xs={12}>
+                <Grid xs={12} className="su-rs-mb-5">
```
``
```suggestion
                <Grid gap xs={12} className="su-rs-mb-5">
```

Tip: We need the gap here so the width of each grid columns is the same as the ones with the grid below 😄 Screenshots showing the reason:
![Screen Shot 2022-07-26 at 11 23 57 PM](https://user-images.githubusercontent.com/42749717/181177214-0133839b-e5f8-4253-a6ba-6e528a152192.png)
![Screen Shot 2022-07-26 at 11 24 08 PM](https://user-images.githubusercontent.com/42749717/181177229-2f66f87b-2304-45c9-bca3-905ca27c18eb.png)

yvonnetangsu commented:
```
                    {body && (
                      <p className="su-subheading su-text-center">{body}</p>
                    )}
```
For line 184 (out of bounds here), if we could add a su-mb-0 for the paragraph that'd be awesome 😄

yvonnetangsu commented on src/components/page-types/registrationFormPage/tripTravelerCard.js:
```diff
@@ -77,39 +77,39 @@ const TripTravelerCard = ({ traveler }) => {
-              <p className="su-basefont-23 su-mb-0">Add traveler</p>
-            </>
-          )}
+          <FlexBox direction="row" alignItems="center" justifyContent="start">
```
A bit of defensive CSS here. If we could add a su-shrink-0 (or su-flex-shrink-0) class to this element, that would prevent the button text from wrapping to 2 lines when the name gets long. I imagine you don't want this 😄 
![Screen Shot 2022-07-26 at 11 37 43 PM](https://user-images.githubusercontent.com/42749717/181178751-21c532e0-5ef5-4c2b-a3bf-f83da45184c3.png)

yvonnetangsu replied:
And while we're here, if we could add a tad more spacing between the icon and "Add traveler" that'd be awesome.

yvonnetangsu commented:
Looking great - thank you for the changes. Left a few more ideas for you 😄

sherakama commented:
> casually including a font weight adjustment for custom journeys and notify me 😬

Lol what?

rebeccahongsf commented:
> > casually including a font weight adjustment for custom journeys and notify me 😬
> 
> Lol what?

For some reason my previous font weight fix for custom journeys and notify me was dropped (most likely during a merge conflict fix)

yvonnetangsu commented on src/components/page-types/registrationFormPage/interstitialPage.js:
```diff
@@ -183,18 +181,26 @@ const InterstitialPage = (props) => {
                   </GridCell>
                   <GridCell xs={12} md={7} lg={8}>
-                    <p className="su-intro-text">
+                    <p className="intro-text su-mb-0 su-rs-mt-1">
```
Where did the su- prefix go my friend 🤣 ?

rebeccahongsf replied:
![](https://c.tenor.com/WnjJsVOwoJQAAAAd/john-travolta-well.gif)

yvonnetangsu commented:
We're super close. Here are my final comments @rebeccahongsf 😄 

1. In general, I try to avoid using `su-grid-gap` for `FlexBox` for now. Even though flex box supports the CSS `gap` property now, but Safari (esp iOS Safari) was pretty late with adopting it, so some versions of Safari v14 still doesn't support the `gap` property 😬 
https://caniuse.com/flexbox-gap
![Screen Shot 2022-07-28 at 3 14 21 PM](https://user-images.githubusercontent.com/42749717/181646464-67c7fdf5-379d-41e3-9a12-4c99181c840d.png)

Technically we only support the last 2 versions of a browser but I try to not cut it too close 😂 

2. Perhaps something to check with Bhavika, it seems like this might look a little better if either a) both heading and the "No travelers have been selected" are center aligned or b) both left aligned 🤔 This looks a little bit off to me but please feel free to push back (totally nitpicky me)
![Screen Shot 2022-07-28 at 3 06 30 PM](https://user-images.githubusercontent.com/42749717/181646967-6bd75bee-433a-4e23-890d-a89b902f3543.png)

4. Could the "Next" button use the existing component `SAALinkButton` similar to how we used it on the Trip page? Are we using the bright red outline variant instead of the solid (seems like the Figma mock uses the outline style)?
https://github.com/SU-SWS/saa_alumni/blob/81688a62e27f80d01df0426ada85be162232b5bb/src/components/page-types/TripPage/TripPageOverviewSection.jsx#L162

4. Are we still displaying the text "You have not added any additional travelers" when one only added themselves?
![Screen Shot 2022-07-28 at 3 26 02 PM](https://user-images.githubusercontent.com/42749717/181647605-bb04620e-7f5e-4af5-b7ea-dfc9f75f42b3.png)

5. Let's use `su-text-18` for the "Remove" button on the right side bar
6. I'd probably want to make sure the spacing above and below the grey separator is the same (might be more obvious when there are more than 2 names

![Screen Shot 2022-07-28 at 3 30 11 PM](https://user-images.githubusercontent.com/42749717/181648033-327a00b4-6427-4575-a080-4efeccf81081.png)

That should be all - thank you so much for the changes Rebecca Hong and for putting up with me @rebeccahongsf 🤣

rebeccahongsf commented:
> 

1. Ahh! That is great to know and duly noted for future usage. Thank you, Yvonne!! (ง’̀-‘́)ง 
2. Bhavika and I are going to discuss this further and make adjustments in a future PR
3. Since we need to use the Gatsby `<Link />` to pass the traveler information, I've left the component as is but updated the styling and icon type to use Hero Icons 😄 
4. This is a great question! Same as #2, Bhavika and I are going to discuss this further and make adjustments (if any) in a future PR.
5. Sounds good! The `remove` text for the traveler list has been updated 👍 
6. Great catch — I've updated the spacing and also added in the button margin-top spacing that was missing before.

I'm going to merge this in so Bhavika and I can review during out next QA theme session 🚀  Thank you again for all your amazing feedback on this, @yvonnetangsu 😸
